packages/coin-tester/src/components/coins/avaxc/index.tsx

```diff
@@ -0,0 +1,280 @@
+      await web3.eth.sendSignedTransaction(signedTx);
+
+      return signedTx;
+    }, setSignedERC20Transaction);
+  };
+
+  const signSmartContractTransaction = async () => {
+    handleState(async () => {
+      const transaction = {
+        nonce: web3.utils.toHex(await web3.eth.getTransactionCount(address, 'pending')),
+        gasPrice: web3.utils.toHex(await web3.eth.getGasPrice()),
+        gasLimit: web3.utils.toHex(8000000),
+        to: '0x256C73C6Dc56710E8BE6B1c3673Caa1B2173fF27',
+        value: '0x0',
+        data: '0x6057361d000000000000000000000000000000000000000000000000000000000000000b',
+      };
+
+      const appId = localStorage.getItem('appId');
+      if (!appId) throw new Error('No Appid stored, please register!');
+
+      const signedTx = await temp.signSmartContractTransaction(transport!, appPrivateKey, appId, 0, transaction);
+
+      await web3.eth.sendSignedTransaction(signedTx);
+
+      return signedTx;
+    }, setSignedSmartContractTransaction);
+  };
+
+  const signMessage = async () => {
+    handleState(async () => {
+      const appId = localStorage.getItem('appId');
+      if (!appId) throw new Error('No Appid stored, please register!');
+
+      const message = 'hellow world';
+
+      const signedMessage = await web3.eth.accounts.sign(message, appPrivateKey);
+
+      const signedTx = await temp.signMessage(transport!, appPrivateKey, appId, 0, web3.utils.toHex(message));
+
+      const recover = await web3.eth.accounts.recover(message, signedTx);
+
+      console.log('recover', recover);
+
+      return signedTx;
+    }, setSignedMessage);
+  };
+
+  const signTypedData = async () => {
+    handleState(async () => {
+      const appId = localStorage.getItem('appId');
+      if (!appId) throw new Error('No Appid stored, please register!');
+
+      const typedData = {
+        types: {
+          EIP712Domain: [
+            { name: 'name', type: 'string' },
+            { name: 'version', type: 'string' },
+            { name: 'chainId', type: 'uint256' },
+            { name: 'verifyingContract', type: 'address' },
+          ],
+          Mail: [
+            { name: 'from', type: 'address' },
+            { name: 'to', type: 'address' },
+            { name: 'value', type: 'uint256' },
+          ],
+        },
+        domain: {
+          name: 'Demo',
+          version: '1.0',
+          chainId: 1,
+          verifyingContract: '0x0000000000000000000000000000000000000000',
+        },
+        primaryType: 'Mail',
+        message: {
+          from: '0xd22b659b218fabf5936a4052f699d373f2b74313',
+          to: '0x0000000000000000000000000000000000000000',
+          value: 12345,
+        },
+      };
+
+      const signedTx = await temp.signTypedData(transport!, appPrivateKey, appId, 0, typedData);
+
+      const recover = sigUtil.recoverTypedSignature_v4({
+        data: typedData,
+        sig: signedTx,
+      });
+
+      console.log('recover', recover);
```

Comment:
Is there any need to add these lines?

Reply:
This section is verifying that if signedTx is correct, recover will be equal to the wallet address